internal/factsengine/gatherers/sbddump.go

```diff
@@ -0,0 +1,147 @@
+package gatherers
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SBDDumpGathererName = "sbd_dump"
+)
+
+// nolint:gochecknoglobals
+var (
+	SBDDevicesLoadingError = entities.FactGatheringError{
+		Type:    "sbd-devices-loading-error",
+		Message: "error loading the configured sbd devices",
+	}
+
+	SBDDumpCommandError = entities.FactGatheringError{
+		Type:    "sbd-dump-command-error",
+		Message: "error while executing sbd dump",
+	}
+)
+
+type SBDDumpGatherer struct {
+	executor    utils.CommandExecutor
+	sbdGatherer *SBDGatherer
+}
+
+func NewDefaultSBDDumpGatherer() *SBDDumpGatherer {
+	return NewSBDDumpGatherer(utils.Executor{}, NewDefaultSBDGatherer())
+}
+
+func NewSBDDumpGatherer(executor utils.CommandExecutor, sbdGatherer *SBDGatherer) *SBDDumpGatherer {
+	return &SBDDumpGatherer{
+		executor:    executor,
+		sbdGatherer: sbdGatherer,
+	}
+}
+
+func (gatherer *SBDDumpGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting %s facts gathering process", SBDDumpGathererName)
+
+	configuredDevices, err := loadDevices(gatherer.sbdGatherer)
+
+	if err != nil {
+		return facts, SBDDevicesLoadingError.Wrap(err.Error())
+	}
+
+	for _, factRequest := range factsRequests {
+		var fact entities.Fact
+
+		if devicesDumps, err := getSBDDevicesDumps(gatherer.executor, configuredDevices); err == nil {
+			fact = entities.NewFactGatheredWithRequest(factRequest, &entities.FactValueMap{Value: devicesDumps})
+		} else {
+			gatheringError := SBDDumpCommandError.Wrap(err.Error())
+
+			fact = entities.NewFactGatheredWithError(factRequest, gatheringError)
+		}
+		facts = append(facts, fact)
+	}
+
+	log.Infof("Requested %s facts gathered", SBDDumpGathererName)
+	return facts, nil
+}
+
+func loadDevices(gatherer *SBDGatherer) ([]string, error) {
+	sbdDevicesRequest := []entities.FactRequest{
+		{
+			Name:     "configured_devices",
+			Gatherer: "sbd_config",
+			Argument: "SBD_DEVICE",
```

Comment:
Didn't consider that :eyes: thanks for bringing up.
I'll se what heppens when adjusting to `cluster.LoadSbdConfig` instead of gatherer composition.

In any case the possibilities are two:
- `SBD_DEVICE` is not even present, which results in a gathering error
- `SBD_DEVICE` is empty which results in an empty list of devices
which of these determines diskless sbd?

Reply:
I went cautious with the [latest changes](https://github.com/trento-project/agent/pull/156/files#diff-d78d11a990a55f93c99ffa880338db0b0e4154502b8c802cbd6620d1b987a7e7R84) in `loadDevices`: if the entry `SBD_DEVICE` is not in the config or it's empty an error is returned, excluding the possibility to use the `sbd_dump` gatherer with diskless sbd.

Would this be enough?